src/views/Pools/components/PoolsTable/ActionPanel/Harvest.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react'
+import { Button, Text, useModal, Flex, TooltipText, useTooltip, Skeleton } from '@pancakeswap/uikit'
+import BigNumber from 'bignumber.js'
+import { useWeb3React } from '@web3-react/core'
+import { convertSharesToCake } from 'views/Pools/helpers'
+import { PoolCategory } from 'config/constants/types'
+import { formatNumber, getBalanceNumber, getFullDisplayBalance } from 'utils/formatBalance'
+import { useTranslation } from 'contexts/Localization'
+import Balance from 'components/Balance'
+import { useCakeVault, useBusdPriceFromToken } from 'state/hooks'
+import { BIG_ZERO } from 'utils/bigNumber'
+import { Pool } from 'state/types'
+
+import { ActionContainer, ActionTitles, ActionContent } from './styles'
+import CollectModal from '../../PoolCard/Modals/CollectModal'
+import UnstakingFeeCountdownRow from '../../CakeVaultCard/UnstakingFeeCountdownRow'
+
+interface HarvestActionProps extends Pool {
+  isAutoVault: boolean
+  userDataLoaded: boolean
+}
+
+const HarvestAction: React.FunctionComponent<HarvestActionProps> = ({
+  sousId,
+  poolCategory,
+  earningToken,
+  userData,
+  userDataLoaded,
+  isAutoVault,
+}) => {
+  const { t } = useTranslation()
+  const { account } = useWeb3React()
+
+  const earnings = userData?.pendingReward ? new BigNumber(userData.pendingReward) : BIG_ZERO
+  const earningTokenPrice = useBusdPriceFromToken(earningToken.symbol)
+  // These will be reassigned later if its Auto CAKE vault
+  let earningTokenBalance = getBalanceNumber(earnings, earningToken.decimals)
+  let earningTokenDollarBalance = getBalanceNumber(earnings.multipliedBy(earningTokenPrice), earningToken.decimals)
+  let hasEarnings = earnings.gt(0)
+  const fullBalance = getFullDisplayBalance(earnings, earningToken.decimals)
+  const formattedBalance = formatNumber(earningTokenBalance, 3, 3)
+  const earningsDollarValue = formatNumber(earningTokenDollarBalance)
+  const isCompoundPool = sousId === 0
+  const isBnbPool = poolCategory === PoolCategory.BINANCE
+
+  // Auto CAKE vault calculations
+  const {
+    userData: { cakeAtLastUserAction, userShares },
+    pricePerFullShare,
+    fees: { performanceFee },
+  } = useCakeVault()
+  const hasAutoEarnings =
+    account && cakeAtLastUserAction && cakeAtLastUserAction.gt(0) && userShares && userShares.gt(0)
+  const { cakeAsBigNumber } = convertSharesToCake(userShares, pricePerFullShare)
+  const cakeProfit = cakeAsBigNumber.minus(cakeAtLastUserAction)
+  const cakeToDisplay = cakeProfit.gte(0) ? getBalanceNumber(cakeProfit, 18) : 0
+  const dollarValueOfCake = cakeProfit.times(earningTokenPrice)
+  const dollarValueToDisplay = dollarValueOfCake.gte(0) ? getBalanceNumber(dollarValueOfCake, 18) : 0
```

Comment:
Obviously we do a lot of this in the card too. I feel like all of my recent comments are on this theme - can we be smarter and share the logic between the table / card view where possible?

Reply:
Reply to this and couple of other comments about copy-paste.
Yeah, I was kinda trying not to mess with cards initially so I don't end up breaking everything. I'm now refactoring/extracting some common stuff where possible. Not everything though since for example prices are getting moved to redux state by you so there is no point making more abstractions based on hooks.